### PR TITLE
Add head_mask and decoder_head_mask to TF LED

### LINKED ISSUE #9814: Missing head_mask and decoder_head_mask arguments in encoder-decoder models
# 🚀 Feature request

Following the PRs #9569, #9634 and #9639, there are other encoder-decoder models, which either do not support `head_mask` and `decoder_head_mask` input arguments at all or can be only provided with a single `head_mask` argument used for head masking both in encoder and decoder. It would be, therefore, nice to make this feature uniform over all the decoder-models.

<hr>

**Models:**

| Model | Pytorch | TensorFlow | PR | Copy dependency |
| ------ | :------: | :---------: | :--: | :-----: |
| BERTGeneration | ☑️  | ✖️  | - | - |
| EncoderDecoderModel | ☑️  | ✖️  | - | - |
| FSMT | ☑️  | ✖️  | #9819 | - |
| LED | ☑️ | ☑️ | PT - #9856 ; TF | - |
| ProphetNet | ☑️  | ✖️  | - | - |
| Longformer | ☑️  | ✖️  | #9856  | LED |

## Your contribution

I'm happy to add this feature in the following days, both for PyTorch and TensorFlow models. (Likely in shorter PRs in order not to create large, overwhelming PRs)

<hr>

Reviewers: @patrickvonplaten, @jplu, @sgugger, @LysandreJik, @stas00 .

### PR DESCRIPTION
**Pull request by stancld**
This PR implements `head_mask` and `decoder_head_mask` for TF LED (and Longformer as there's a copy dependency) and it is the follow-up to the open issue #9814. 

**Motivation:** This PR is a part of an endeavour to enable the usage of `head_mask` and `decoder_head_mask` for all encoder-decoder transformers following the recent work on BART-like models (#9639).

<hr>

Fixes: https://github.com/huggingface/transformers/issues/9814

Reviewers: @jplu @patrickvonplaten @LysandreJik @sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
LGTM, thanks for adding this!

File: src/transformers/models/led/modeling_tf_led.py [hunk 1]
```diff
@@ -817,6 +828,20 @@ def _compute_global_attn_output_from_hidden(
         # compute global attn probs
         global_attn_probs_float = tf.nn.softmax(global_attn_scores, axis=-1)
 
+        # apply layer head maskin
```
Comment (patrickvonplaten) [inline src/transformers/models/led/modeling_tf_led.py:831]:
```suggestion
        # apply layer head masking
```

**Review by patrickvonplaten — APPROVED** [review]
Looks great! Awesome that you've added the head masking also for both the local and global attention. Longformer is a tricky model so great job!

**Review by jplu — APPROVED** [review]
LGTM! Thanks a lot for the great work on this!

**Review by LysandreJik — APPROVED** [review]
Great, thanks @stancld!
